### ISSUE TITLE
Update Huey task queue library

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2731]
We have a number of usefull fixes from when we initially introduced (and pinned) Huey to 2.3.0. It is proposed that we update this pinning or adopt a major version only pin instead.
PyPi: https://pypi.org/project/huey/#history

Noteworthy additions from the upstream changelog: https://github.com/coleifer/huey/blob/master/CHANGELOG.md

- 2.5.0 Fix bug in SIGINT and SIGTERM behavior for gevent users.
- Add an on_commit_task() decorator for Django extension that will enqueue the task after any database changes have been committed. This eliminates a common race condition where a task is enqueued and executed before the corresponding database changes have been committed.
- Check to ensure the gevent monkeypatch was applied when running the consumer with greenlet workers, log warning if it is not.
- 2.4.1 Attempt to reconnect to database if connection becomes unusable (e.g. due to a server restart). See: huey.contrib.sql_huey.SqlHuey.
- 2.4.0 Task expiration: https://huey.readthedocs.io/en/latest/guide.html#task-expiration - we have had reports of task ongoing  indefinitely that may benefit here.

We also have a number of other bug-fixes, and Django enhancements in the interim releases.

**Comment by phillxnet** [comment]
By removing the pinning, at least for now, we have:
```
lbuildvm:/opt/rockstor # poetry update
Updating dependencies
Resolving dependencies... (10.0s)

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating huey (2.3.0 -> 2.5.0)

```

**Comment by phillxnet** [comment]
# Testing
## Balance
```
[11/Nov/2023 17:28:57] INFO [storageadmin.tasks:64] Task [start_balance], id: edc582b8-22e3-4ff8-98dd-840ef999a190 completed OK
[11/Nov/2023 17:28:57] INFO [storageadmin.tasks:67] Updating end_time accordingly to 2023-11-11 17:28:57.093087+00:00
[11/Nov/2023 17:28:57] INFO [storageadmin.tasks:56] Now executing Huey task [update_end_time], id: bca8a0af-ecdd-4e00-99cc-2be52b8aa360.
[11/Nov/2023 17:28:57] INFO [storageadmin.tasks:64] Task [update_end_time], id: bca8a0af-ecdd-4e00-99cc-2be52b8aa360 completed OK

```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2741
